core/bindings.rs

```diff
@@ -361,9 +408,13 @@ fn opcall<'s>(
   };
   let op = OpTable::route_op(op_id, state.op_state.clone(), payload);
   match op {
-    Op::Sync(result) => {
-      rv.set(result.to_v8(scope).unwrap());
-    }
+    Op::Sync(result) => match result {
+      OpResult::Ok(_) => throw_type_error(
+        scope,
+        format!("Can not call a sync op [{}] with opAsync()", op_id),
+      ),
+      OpResult::Err(_) => rv.set(result.to_v8(scope).unwrap()),
```

Comment:
Why does this path still need to work?

Reply:
That's how serde_v8 deserialization errors are currently returned, that happens sync since we deserializing before jumping into the async fn

Reply:
Are these thrown synchronously from opAsync, or are they returned via the promise?

Reply:
These are thrown synchronously, FYI this is the current behavior this PR doesn't change it, it was implicitly handled by the sync branch when both were mangled